src/BornOnThisDay.java

```diff
@@ -0,0 +1,149 @@
+}
+    /**
+     * BornOnThisDay.convertAndFilterResponse() parses and filters the return value of getResponse()
+     * to produce a new JSON Array to be used to render the front end.
+     *
+     * @param response the return value of getResponse()
+     *
+     * @return JSONArray of objects with the following key-value pairs:
+     * "name"-name of the person, "year"-the year of their birth, "tagline"--a phrase describing the person,
+     * "description" - a 2-3 sentence description of the person, "imageUrl"-the url of a thumbnail image of the person
+     *
+     * @author Bradley Boggs, bradleywboggs@gmail.com
+     */
+
+    public static JSONArray convertAndFilterResponse(String response) {
+        try {
+            JSONObject initJSONObject = new JSONObject(response);
+            JSONArray initArray = (JSONArray) initJSONObject.get("births");
+            JSONArray returnArray = new JSONArray();
+
+            for (int i = 0; i < initArray.length(); i++) {
+                JSONObject returnArrayElement = new JSONObject();
+                JSONObject arrayElement = (JSONObject) initArray.opt(i);
+
+                if (arrayElement.has("year")) {
+                    Integer year = (Integer) arrayElement.opt("year");
+                    returnArrayElement.put("year", year);
+                }
+
+                JSONArray pagesArray = (JSONArray) arrayElement.opt("pages");
+                for (int j = 0; j < pagesArray.length(); j++) {
+                    JSONObject pagesElement = (JSONObject) pagesArray.opt(j);
+
+                    if (pagesElement.has("extract")) {
+                        String description = (String) pagesElement.opt("extract");
+                        returnArrayElement.put("description", description);
+                    }
+
+                    if (pagesElement.has("description")) {
+                        String tagline = (String) pagesElement.opt("description");
+                        returnArrayElement.put("tagline", tagline);
+                    }
+
+                    if (pagesElement.has("titles")) {
+                        JSONObject titlesObject = (JSONObject) pagesElement.opt("titles");
+                        if (titlesObject.has("display")) {
+                            String name = (String) titlesObject.opt("display");
+                            returnArrayElement.put("name", name);
+                        }
+                    }
+
+                    if (pagesElement.has("thumbnail")) {
+                        JSONObject thumbnailObject = (JSONObject) pagesElement.opt("thumbnail");
+                        String imageUrl = (String) thumbnailObject.opt("source");
+                        returnArrayElement.put("imageUrl", imageUrl);
+                    }
+
+
+                }
+
+                returnArray.put(returnArrayElement);
+
+            }
+            return returnArray;
+        } catch (JSONException e) {
+            e.printStackTrace();
+            return null;
```

Comment:
I don't know if this is the best choice to return when there's an error. Maybe an empty JSON list? Or an explicit http error so I can handle it on the front end properly.